### PR TITLE
Add zstash ls warning

### LINKED ISSUE #100: zstash reads stored index.db instead of retrieving the new one specified by --hpss
Hi zstash team,

I tried to list files from two different HPSS locations.
It seems zstash uses the first index.db even though I specified a new location using --hpss.
The command I use is `zstash ls --hpss=[location 1/2]`.

How can I avoid this problem?
I found it a little bit dangerous for new users.

Thank you!

### PR DESCRIPTION
**Pull request by forsyth2**
Add warning about how `zstash ls` works in relation to its cache. Resolves #100.

### CODE REVIEW

**Comment by forsyth2** [comment]
@golaz I added a warning box to the docs for `zstash ls`. Do you think that adequately covers the issue? The only file to review is `source/usage.rst`.

@chengzhuzhang mentioned on #100 that it might be helpful to include a warning (on the `master` branch code) regarding overwriting `index.db`. Do you think we should do that too to properly resolve #100?

**Comment by chengzhuzhang** [comment]
> @chengzhuzhang mentioned on #100 that it might be helpful to include a warning (on the `master` branch code) regarding overwriting `index.db`. Do you think we should do that too to properly resolve #100?

I think to add a note in documentation about the default behavior that zstash will only retrieve index.db from source hpss directory if there is not a local cache present. And add the tip suggested by Chris: "if you change --hpss between calls to zstash, either delete the zstash cache sub-directory or execute zstash from another sub-directory".  For code change, it might be helpful to include the time stamp when the index.db file being extracted?

File: docs/html/_sources/usage.rst.txt [hunk 1]
```diff
@@ -348,6 +348,12 @@ where
     to avoid shell substitution.
   * Names of specific tar archives to list all files within these tar archives.
 
+.. warning::
+    Running ``zstash ls`` creates a cahce (by default named ``zstash``).
```
Comment (golaz) [inline docs/html/_sources/usage.rst.txt:352]: "cahce" is mispelled here, but spelled correctly in the rst file below. You probably need to rerun 'make html'

**Review by golaz — APPROVED** [review]

File: source/usage.rst [hunk 2]
```diff
@@ -358,6 +358,31 @@ Below is an example. Note the names of the columns:  ::
    30482	archive/logs/atm.log.8229335.180130-143234.gz	20156521	2018-02-01 10:02:35	e8161bba53500848dc917258d1d8f56a	000018.tar	131697281536	
    51608	case_scripts/logs/atm.log.8229335.180130-143234.gz	20156521	2018-02-01 10:02:52	e8161bba53500848dc917258d1d8f56a	000047.tar	202381473280	
 
+.. warning::
+    Running ``zstash ls`` creates a cache (by default named ``zstash``).
+    ``zstash`` will only retrieve ``index.db`` from the HPSS archive
+    if this local cache is not present.
+
+Example 1 -- updating the HPSS archive: ::
```
Comment (forsyth2) [inline source/usage.rst:366]: Confirm examples work once Cori is back online.

**Review by chengzhuzhang — APPROVED** [review]

**Comment by forsyth2** [comment]
@chengzhuzhang @golaz I made substantial changes from previous commits, listing 4 examples of the warning. I wanted to make sure I understood what was going on and I think the examples would be useful to others. See `source/usage.rst` to review. Thanks!

**Comment by golaz** [comment]
@forsyth2 : the additional examples you added are fine. Let's hope they help clarify some user questions. You can go ahead and merge the changes.

**Pull request merged**